lua/neogit/status.lua

```diff
@@ -64,390 +69,191 @@ local function get_section_item_for_line(linenr)
     return nil, nil
   end
 
-  return section, status[section.name][item_idx]
+  return section, section.files[item_idx]
 end
 
 local function get_current_section_item()
   return get_section_item_for_line(vim.fn.line("."))
 end
 
-local function toggle_sign_at_line(line)
-  local sign_info = status_buffer:get_sign_at_line(line, "fold_markers")
-  local sign = sign_info.signs[1]
-
-  if sign ~= nil then
-    local parts = vim.split(sign.name, ":")
-    local new_name = (parts[1] == "NeogitOpen" and "NeogitClosed" or "NeogitOpen") .. ":" .. parts[2]
-    status_buffer:place_sign(line, new_name, 'fold_markers')
-  end
-end
-
-local function toggle()
-  vim.cmd("silent! normal za")
-
-  if not config.values.disable_signs then
-    local section, item = get_current_section_item()
-
-    if section == nil then
-      return
-    end
-
-    local line = item ~= nil and item.first or section.first
-
-    local on_hunk = item ~= nil and line_is_hunk(vim.fn.getline('.'))
-
-
-    if on_hunk then
-      local ignored_sections = { "untracked_files", "stashes", "unpulled", "unmerged" }
-
-      for _, val in pairs(ignored_sections) do
-        if val == section.name then
-          return
-        end
+local function new_output()
+  return setmetatable({ }, {
+    __index = {
+      append = function (tbl, data)
+        if type(data) == 'string' then table.insert(tbl, data)
+        elseif type(data) == 'table' then
+          for _, r in ipairs(data) do table.insert(tbl, r) end
+        else error('invalid data type: ' .. type(data), 2) end
+        return tbl
+      end,
+      into_buffer = function (tbl, from, to)
```

Comment:
When I read something with `into....` I always think about a conversion and not a side effect. Could it be possible to change this name?

Reply:
After thinking about this some more I don't really think that `into_buffer` should be a part of this class (I guess we can this lua metatable magic classes?).

Reply:
Yeah that thing is a mess :sweat_smile: Currently it's just a helper decorator that wraps the lines i felt were too verbose; it can't really exist outside of the context it's used right now as well, because it closes around the status buffer.
As for the `into_buffer` method, how does a `replace_content_with`-method on the buffer object sound to you instead? :)